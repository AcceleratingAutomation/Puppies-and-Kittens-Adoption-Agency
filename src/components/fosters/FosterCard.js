import React from "react";
import { Grid, Typography } from "@material-ui/core";
import "../../styles.css";
import { SummaryCard } from "../SummaryCard";

const FosterCard = ({ displayName, id, email, numCurrentRescues, numTotalRescues, image, isAccepting }) => {
  const foster = { displayName, id, email, numCurrentRescues, numTotalRescues, image, isAccepting };

  return (
    <SummaryCard type={"foster"} image={foster.image} name={foster.displayName} viewComponentDetailsUrl={`/v1/foster/${foster.id}`} >
      <Grid item xs={12}>
        <Typography variant="h4">{foster.displayName}</Typography>
      </Grid>
      <Typography variant="h5" gutterBottom>
        {foster.email}
      </Typography>
      <Typography variant="h5" gutterBottom>
        Accepting Rescues: {foster.isAccepting ? "Yes" : "No"}
      </Typography>
      <Typography variant="h5" gutterBottom>
        Current Rescues: {foster.numCurrentRescues}
      </Typography>
      <Typography variant="h5" gutterBottom>
        Total Rescues: {foster.numTotalRescues}
      </Typography>
      <Grid item xs={12} container justify="center">
      </Grid>
    </SummaryCard>
  );
};

export default FosterCard;