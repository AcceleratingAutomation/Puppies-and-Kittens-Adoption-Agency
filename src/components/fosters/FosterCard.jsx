import React from "react";
import { Grid, Typography } from "@material-ui/core";
import "../../styles.css";
import PropTypes from "prop-types";
import SummaryCard from "../SummaryCard";
import { fostersEndpoint } from "../../server/apiService/apiConfig";

function FosterCard({
  name,
  id,
  email,
  numCurrentRescues,
  numTotalRescues,
  image,
  isAccepting,
}) {
  const foster = {
    name,
    id,
    email,
    numCurrentRescues,
    numTotalRescues,
    image,
    isAccepting,
  };

  return (
    <SummaryCard
      directory="users"
      type="foster"
      id={foster.id}
      image={foster.image}
      name={foster.name}
      viewComponentDetailsUrl={`${fostersEndpoint}/${foster.id}`}
    >
      <Grid item xs={12}>
        <Typography variant="h4">{foster.name}</Typography>
      </Grid>
      <Typography variant="h5" gutterBottom>
        {foster.email}
      </Typography>
      <Typography variant="h5" gutterBottom>
        Accepting Rescues: {foster.isAccepting ? "Yes" : "No"}
      </Typography>
      <Typography variant="h5" gutterBottom>
        Current Rescues: {foster.numCurrentRescues}
      </Typography>
      <Typography variant="h5" gutterBottom>
        Total Rescues: {foster.numTotalRescues}
      </Typography>
      <Grid item xs={12} container justify="center" />
    </SummaryCard>
  );
}

FosterCard.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  numCurrentRescues: PropTypes.number.isRequired,
  numTotalRescues: PropTypes.number.isRequired,
  image: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  isAccepting: PropTypes.bool.isRequired,
};

export default FosterCard;
