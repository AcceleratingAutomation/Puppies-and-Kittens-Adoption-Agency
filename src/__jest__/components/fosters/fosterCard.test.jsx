import React from "react";
import { render, screen } from "@testing-library/react";
import FosterCard from "../../../components/fosters/FosterCard";

function renderFosterCard(props) {
  return render(
    <FosterCard
      id={props.id}
      name={props.name}
      email={props.email}
      numCurrentRescues={props.numCurrentRescues}
      numTotalRescues={props.numTotalRescues}
      image={props.image}
      isAccepting={props.isAccepting}
    />,
  );
}

describe("Foster Card", () => {
  const fosterProps = {
    id: "1",
    name: "John Doe",
    email: "john.doe@example.com",
    numCurrentRescues: 5,
    numTotalRescues: 10,
    image: "thumbnail-image",
    isAccepting: true,
  };

  beforeEach(() => {
    renderFosterCard(fosterProps);
  });

  test("matches snapshot comparison", () => {
    const { asFragment } = renderFosterCard(fosterProps);
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays foster name", () => {
    expect(screen.getByText(fosterProps.name)).toBeInTheDocument();
  });

  test("displays email", () => {
    expect(screen.getByText(fosterProps.email)).toBeInTheDocument();
  });

  test("displays number of current rescues", () => {
    expect(
      screen.getByText(`Current Rescues: ${fosterProps.numCurrentRescues}`),
    ).toBeInTheDocument();
  });

  test("displays number of total rescues", () => {
    expect(
      screen.getByText(`Total Rescues: ${fosterProps.numTotalRescues}`),
    ).toBeInTheDocument();
  });

  test("displays accepting rescues", () => {
    expect(
      screen.getByText(
        `Accepting Rescues: ${fosterProps.isAccepting ? "Yes" : "No"}`,
      ),
    ).toBeInTheDocument();
  });

  test("displays not accepting rescues", () => {
    const nonAcceptingProps = {
      ...fosterProps,
      isAccepting: false,
    };
    renderFosterCard(nonAcceptingProps);
    expect(screen.getByText("Accepting Rescues: No")).toBeInTheDocument();
  });
});
