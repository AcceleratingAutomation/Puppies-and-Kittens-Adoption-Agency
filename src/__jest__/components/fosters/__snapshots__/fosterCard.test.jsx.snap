// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foster Card matches snapshot comparison 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 summary-card css-1mbunh-MuiPaper-root"
  >
    <img
      alt="John Doe's picture"
      class="responsive-image"
      data-testid="display-image"
      src="/images/users/thumbnail-image.webp"
      style="border-radius: 25%; width: 15rem; height: 18rem; object-fit: cover; margin: 2vh;"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          John Doe
        </h4>
      </div>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        john.doe@example.com
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Accepting Rescues: Yes
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Current Rescues: 5
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Total Rescues: 10
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-s2k0j8-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-s2k0j8-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary summary-card-button css-11qr2p8-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          View Details
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
